### PR TITLE
#10370 metadata search in comments

### PR DESCRIPTION
**Pull request by flambchap**
Currently implemented for PostgreSQL, SQL Server, Oracle, Exasol, and MySQL.

### CODE REVIEW

**Review by LonwoLonwo — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBSStructureAssistant.java [hunk 1]
```diff
@@ -39,28 +37,94 @@
...
+     *
+     * These include:
+     * <ul>
+     *     <li>parentObject: parent (schema or catalog)</li>
+     *     <li>objectTypes: type of objects to search</li>
+     *     <li>mask: name mask</li>
+     *     <li>caseSensitive: case sensitive search (ignored by some implementations)</li>
+     *     <li>globalSearch: search in all available schemas/catalogs. If {@code false} then search with respect of active schema/catalog</li>
+     *     <li>maxResults: maximum number of results</li>
+     *     <li>searchInComments: perform additional search in comments (ignored by some implementations)</li>
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBSStructureAssistant.java]: Consider surrounding parameter names with `<code>name here</code>` for better visibility

Comment (flambchap) [reply plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBSStructureAssistant.java]: Good point

**Review by ShadelessFox — APPROVED** [review]
Looks fine to me :)

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/dialogs/GotoObjectDialog.java [hunk 2]
```diff
@@ -84,15 +87,39 @@ public GotoObjectDialog(Shell shell, DBCExecutionContext context, DBSObject cont
...
+        Composite composite = UIUtils.createComposite(parent, 1);
+        composite.setLayoutData(new GridData(GridData.FILL_BOTH));
+
+        createObjectTypesGroup(composite);
+
+        Button searchInCommentsCheckbox = UIUtils.createCheckbox(
+                composite,
+                UINavigatorMessages.dialog_search_objects_search_in_comments,
+                !doNotSearchInComments
+        );
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/navigator/dialogs/GotoObjectDialog.java]: We use 4-space indentation for continuations.

You can configure code style for Intellij IDEA in <kbd>Editor</kbd> &rArr; <kbd>Code Style</kbd> &rArr; <kbd>Java</kbd> &rArr; <kbd>Continuation indent</kbd>.

File: plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/internal/UINavigatorMessages.java [hunk 3]
```diff
@@ -229,6 +229,7 @@
     public static String question_no_sql_available;
     public static String error_sql_generation_title;
     public static String error_sql_generation_message;
+    public static String dialog_search_objects_search_in_comments;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/internal/UINavigatorMessages.java]: Consider using `dialog_project_goto_object_` prefix for `GotoObjectDialog`-related localization strings and move it here:
https://github.com/dbeaver/dbeaver/blob/0f8c378219dcf94f6dc5643772219f9dcf2cf60e/plugins/org.jkiss.dbeaver.ui.navigator/src/org/jkiss/dbeaver/ui/internal/UINavigatorMessages.java#L80-L81

**Review by ShadelessFox — APPROVED** [review]
Nice!

**Review by LonwoLonwo — APPROVED** [review]
Added rows more than deleted. You did it.

**Pull request merged**